src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1027,6 +1027,20 @@ fn link_args<'a, B: ArchiveBuilder<'a>>(cmd: &mut dyn Linker,
     let t = &sess.target.target;
 
     cmd.include_path(&fix_windows_verbatim_for_gcc(&lib_path));
+
+    if t.linker_flavor == LinkerFlavor::Msvc && t.target_vendor == "uwp" {
+        let link_tool = windows_registry::find_tool("x86_64-pc-windows-msvc", "link.exe")
+            .expect("no path found for link.exe");
```

Comment:
This should report errors as first-class errors instead of panics, but for this case I think that all errors should be swallowed and this logic ignored. This could perhaps `warn!` on errors but it's probably not worthwhile to halt compilation if this fails.